cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AbstractMonitorHandler.java

```diff
@@ -242,9 +248,9 @@ private void restartInstances(HttpResponder responder, String serviceName, int i
       LOG.debug(String.format("IllegalStateException when trying to restart instances for service %s", serviceName));
```

Comment:
Better to have twill throw a better, checked exception, so you don't have to remember to catch an ``IllegalStateException`` everywhere. Can you file a jira?